lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -3743,15 +3729,6 @@ ResolvedType GenericSignatureBuilder::maybeResolveEquivalenceClass(
     if (!nestedPA)
       return ResolvedType::forUnresolved(baseEquivClass);
 
-    if (resolutionKind != ArchetypeResolutionKind::AlreadyKnown) {
-      // Update for all of the concrete decls with this name, which will
-      // introduce various same-type constraints.
-      for (auto concreteDecl : concreteDecls) {
-        (void)basePA->updateNestedTypeForConformance(*this, concreteDecl,
-                                                     resolutionKind);
```

Comment:
Huh. I'm surprised we can get rid of this. It's supposed to establish equivalences between same-named nested types.

Reply:
To be honest I'm just not sure what it's going to do now because we substitute early which means that we'd end up with new potential archetype that has associated type so we don't really need to add new same-type requirement like before where we had potential archetype with concrete decl and substituted typealias type. But I might be totally missing something here, could you please provide an example?

Reply:
Huh. I had convinced myself that this mattered due to https://github.com/apple/swift/pull/14974, but my change there didn't fix the problem at all. Maybe this is indeed dead code!

Reply:
I suspect the deletion of this loop is the cause of the regression in `./validation-test/compiler_crashers_2_fixed/0075-rdar30248571.swift`.